### PR TITLE
Use gdal to produce raster thumbnails

### PR DESCRIPTION
**Pull request by nirvn**
- PDF, unlocked
- JP2, unlocked

:heart:

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 92625886ff417b78777f089e0684015dbf6e16ac:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2950-8ab62ec4302234d917c704ec6859ad0712464beb-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2950-8ab62ec4302234d917c704ec6859ad0712464beb-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2950-8ab62ec4302234d917c704ec6859ad0712464beb-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2950-8ab62ec4302234d917c704ec6859ad0712464beb-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for bf707c6ae8080157f3081efd8a26a175a6671e47:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2950-9112f10ff7133f54d5e996d7cc7b9f8cb03dfd58-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2950-9112f10ff7133f54d5e996d7cc7b9f8cb03dfd58-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2950-9112f10ff7133f54d5e996d7cc7b9f8cb03dfd58-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2950-9112f10ff7133f54d5e996d7cc7b9f8cb03dfd58-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for f7eb538627ed2bcde91ce8810f001b8bcf9b807a:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2950-dd14ed0d69c7b57a96e7918276cac6eac2995e74-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2950-dd14ed0d69c7b57a96e7918276cac6eac2995e74-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2950-dd14ed0d69c7b57a96e7918276cac6eac2995e74-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2950-dd14ed0d69c7b57a96e7918276cac6eac2995e74-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 12e721678f91a88f6a24eff345cfa7a7d797441f:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2950-2c26158ff73c4336a9449321ea1e62b495ebf745-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2950-2c26158ff73c4336a9449321ea1e62b495ebf745-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2950-2c26158ff73c4336a9449321ea1e62b495ebf745-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2950-2c26158ff73c4336a9449321ea1e62b495ebf745-x86-android.apk)

File: src/core/localfilesimageprovider.cpp [hunk 1]
```diff
@@ -13,21 +13,52 @@
...
+  int bands = std::min( 4, GDALGetRasterCount( dataset.get() ) );
+  if ( bands == 2 )
+  {
+    // For 2-band raster, go for a 1-band grayscale representation
+    bands = 1;
+  }
+
+  const QSize outputSize( requestedSize.width(), rows * requestedSize.width() / cols );
+  QImage image( outputSize, bands == 4 ? QImage::Format_RGBA8888 : bands == 3 ? QImage::Format_RGB888
+                                                                              : QImage::Format_Grayscale8 );
```
Comment (suricactus) [inline src/core/localfilesimageprovider.cpp:46]: poor astyle :(

**Review by suricactus — APPROVED** [review]

**Pull request merged**